src/main/java/uk/gov/hmcts/sscs/service/SubmitAppealService.java

```diff
@@ -59,38 +55,51 @@
     }
 
     public void submitAppeal(SyaCaseWrapper appeal) {
-        String appellantLastName = appeal.getAppellant().getLastName();
-        String nino = appeal.getAppellant().getNino();
-
-        String appellantUniqueId = String.format(ID_FORMAT, appellantLastName, nino.substring(nino.length() - 3));
+        CaseData caseData = transformAppealToCaseData(appeal);
+        CaseDetails caseDetails = createCaseInCcd(caseData);
+        byte[] pdf = generatePdf(appeal, caseDetails.getId());
```

Comment:
A null check needs to be done for caseDetails.getId() if there is any error while storing case to CCD.